.github/workflows/publish_website.yml

```diff
@@ -44,30 +44,25 @@ jobs:
         with:
           python-version: 3.x
           cache: "pip"
-      - id: "gcp-auth"
-        name: "Authenticating to Google Cloud"
-        uses: "google-github-actions/auth@v1"
-        with:
-          token_format: "access_token"
-          credentials_json: "${{ secrets.IREE_OSS_GITHUB_RUNNER_BASIC_TRUST_SERVICE_ACCOUNT_KEY }}"
-          create_credentials_file: false
       - name: Installing dependencies
         run: |
           pip install -r docs/website/requirements.txt
           pip install requests
+          sudo apt update
+          sudo apt install -y ninja-build
+          echo "CC=clang" >> $GITHUB_ENV
+          echo "CXX=clang++" >> $GITHUB_ENV
       - name: Generating release index
         run: |
           ./build_tools/scripts/generate_release_index.py \
             --repo="${GITHUB_REPOSITORY}" \
             --output=docs/website/docs/pip-release-links.html
+      - name: ccache
+        uses: hendrikmuhs/ccache-action@v1.2
```

Comment:
The reason we don't use this in the other jobs we are trying to move is because this cache doesn't provide enough space to replace the cpu build gcp cache jobs right?

Reply:
Yeah. This just builds the documentation files, which needs llvm-tblgen. The cache size ends up being around 5mb for that very small slice of the project. The build of the full compiler and runtime needs somewhere between 300MB and 2GB+. GitHub's cache takes multiple minutes to upload/download 300MB and has a limit of 10GB total across the whole repository, which we'd constantly be thrashing against if we tried to use 2GB cache entries.

![image](https://github.com/user-attachments/assets/87bf7bb6-4752-442e-8ad1-5b35539037f2)
https://github.com/iree-org/iree/actions/caches

Reply:
Without thinking about when to potentially rebuild, could it be useful to build `llvm-tblgen` in a separate workflow and provide it as an artifact that can be downloaded?

Reply:
Eh... maybe. Could get it from a [python] package somewhere. It's pretty fast to build even on a GitHub hosted runner with a warm cache, at least.